### ISSUE TITLE
[Bug] Typeid to Type migrations

### ISSUE THREAD

**Issue by s-diez** [aimeos/aimeos-core#152]
Hi

Typo3: 8.7.24
Aimeos: 2019.04.3 was 2018.10.15

I'm trying to upgrade from 2018.10 to 2019.04. Even after executing the setup script all my images, texts, ... assigned to products do not have a type assigned. I would guess this should have been done through migrations in `\Aimeos\MW\Setup\Task\TypesMigrateColumns`.

As an example lets take "db-media". All my entries in the table "mshop_media" do have a valid "typeid" but they all have "" as "type". Is there a reason there are no entries for the table "mshop_media", "mshop_text", ... in  `\Aimeos\MW\Setup\Task\TypesMigrateColumns->$migrations`? 

https://github.com/aimeos/aimeos-core/blob/e9258bc5758f501bc5716e3dc9596ed25c90d223/lib/mshoplib/setup/TypesMigrateColumns.php#L60-L63

I would have expected an entry like the following:

```php
'mshop_media' => 'UPDATE "mshop_media" SET "type" = ( SELECT "code" FROM "mshop_media_type" AS t WHERE t."id" = "typeid" AND t."domain" = "domain" LIMIT 1 ) WHERE "type" IS NULL',
```

Furthermore is `WHERE "type" IS NULL` actually correct? My db schema for mshop_media_list does not allow type to be null therefore all entries start with value "".

**Comment by aimeos** [comment]
Looks like a bug due to the `NOT NULL` column property, so it should be `WHERE "type" IS NULL` like you say. Can you test and create a PR if that fixes the upgrade process?

**Comment by s-diez** [comment]
Do you mean to just revert https://github.com/aimeos/aimeos-core/commit/e7c2ebccfa666760293abd0f79782eb0fa6aa3f7? This did not resolve the problem, but I think it is still needed.

To me there seem to be two problems. First the check for null on a non null column and second the missing migrations for some tables.

In `\Aimeos\MW\Setup\Task\TypesMigrateColumns->$migrations` there is no entry for the table `mshop_media`, `mshop_text`, ...
As far as I understand the database schema these tables need to be migrated as they had a column 'typeid' and now have a column 'type'.

**Comment by aimeos** [comment]
Yes, don't know why https://github.com/aimeos/aimeos-core/commit/e7c2ebccfa666760293abd0f79782eb0fa6aa3f7 was added.

These tables seem to be missing for migration:
- mshop_media
- mshop_product
- mshop_price
- mshop_service
- mshop_text

**Comment by s-diez** [comment]
Thanks for the feedback.
My local data seems to be migrated now. As we do not use 'mshop_service' its migration is untested.

https://github.com/aimeos/aimeos-core/pull/154

**Comment by aimeos** [comment]
merged
